docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+        memo : opt nat64;
+        amount_e8s : nat64;
+    };
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "Lorem ipsum";
+        url = "Lorem ipsum";
+        summary = "Lorem ipsum";
+        action = opt variant {
+            TransferSnsTreasuryFunds = record {
+                from_treasury = int32;
+                to_principal = opt principal;
+                to_subaccount = opt record { subaccount = vec nat8 };
+                memo = opt nat64;
+                amount_e8s = nat64;
+               
+            };
+        };
+    };
+)'
+```
+
+## `UpgradeSnsControlledCanister` 
+
+### Relevant type signatures
+
+```candid
+    type UpgradeSnsControlledCanister : UpgradeSnsControlledCanister;
+
+    type UpgradeSnsControlledCanister = record {
+        new_canister_wasm : vec nat8;
+        mode : opt int32;
+        canister_id : opt principal;
+        canister_upgrade_arg : opt vec nat8;
+    };
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            UpgradeSnsControlledCanister = record {
+                new_canister_wasm = vec nat8;
+                mode = opt int32;
+                canister_id = opt principal;
+                canister_upgrade_arg = opt vec nat8;
+            };
+        };
+    };
+)'
+```
+
+## `DeregisterDappCanisters` 
+
+### Relevant type signatures
+
+```candid
+    type DeregisterDappCanisters : DeregisterDappCanisters;
+
+    type DeregisterDappCanisters = record {
+        canister_ids : vec principal;
+        new_controllers : vec principal;
+    };
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            DeregisterDappCanisters = record {
+                canister_ids = vec principal;
+                new_controllers = vec principal;
+            };
+        };
+    };
+)'
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            <PROPOSAL_TYPE> = <PARAMETERS_OF_PROPOSAL_TYPE>
+        };
+    }
+)'
+```
+
+## `ManageSnsMetadata` 
+
+### Relevant type signatures
+
+```candid
+    type  ManageSnsMetadata : ManageSnsMetadata;
+
+    type ManageSnsMetadata = record {
+        url : opt text;
+        logo : opt text;
+        name : opt text;
+        description : opt text;
+    };
+```
+
+### Putting it together
+
+Example in bash:
```

Comment:
would make a concrete example here and say "before the proposal it was like this..." 
the proposal now only cahnges 2 of the 4 attributes to this...
To do so, you could send this proposal..

(then it is clear how to define also the things that one does not want to change)

Reply:
Sorry, I am not sure I follow

Reply:
sorry if this was unclear.
so this is a proposal to change the metadata of an SNS.
I thought it might be useful to talk the reader through a concrete example and say "let's say today the metadata is like this (e.g., state a name and description and url and logo). Then, we can make an example and say "let's say you now want to change the description, but leave the url, logo, and name as they are" and then show a concrete proposal that would change only this one field. 
My intent: sometimes it is also hard to know how to specify which fields you _don't_ want to change: do I leave them blank? do I need to just repeat the same value that is already there to say "I want to keep them"? => therefore I think an example that includes both things that an SNS wants to change and things that an SNS does not want to change might be super helpful

Reply:
done